### ISSUE TITLE
Focus does not move to another window

### ISSUE THREAD

**Issue by smpallen99** [zed-industries/zed#8735]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I have 2 zed windows open, lets call them (zed-a) and (zed-b).  After woring in zed-a, I click into zed-b and start typing. Nothing happens in zed-b, but the characters show up in zed-a.  I click into zed-a, hit escape (yes, I'm using vim bindings), go back to zed-b and the problem continues. When I click into either zed window, the app is selected as indicated by color of the OSX window buttons. 

If I got to another app, I can type as expected. 

As I'm typing this and testing, I managed to get zed-b working after hitting `:` to open the command pallete. Now I can enter text in zed-b, but when I click into zed-a I'm having a new issue. I can move around in the file with movement keys, but typing `o` to instert on the next line, nothing happens.  Tried that a couple times without success. Then I tried a `A` and that worked. Now zed-b is working as expected. After that, I went back to zed-a and when I type, the input is going to zed-b. So the problem reproduced in the opposite windows.

### Environment

Zed: v0.125.0 (Zed Preview)
OS: macOS 13.3.1
Memory: 32 GiB
Architecture: x86_64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

Nothing interesting in Zed.log...

```
2024-03-02T19:09:20-05:00 [INFO] Language server with id 2 sent unhandled notification $/status/report:
{
  "text": "😺Lua",
  "tooltip": "Workspace   : /Users/spallen/.config/nvim-Modular\nCached files: 47/33\nMemory usage: 8M"
}
2024-03-02T19:09:23-05:00 [INFO] Language server with id 2 sent unhandled notification $/status/report:
{
  "text": "😺Lua",
  "tooltip": "Workspace   : /Users/spallen/.config/nvim-Modular\nCached files: 48/33\nMemory usage: 8M"
}
2024-03-02T19:09:24-05:00 [INFO] Language server with id 2 sent unhandled notification $/status/report:
{
  "text": "😺Lua",
  "tooltip": "Workspace   : /Users/spallen/.config/nvim-Modular\nCached files: 48/33\nMemory usage: 9M"
}
2024-03-02T19:09:28-05:00 [INFO] Language server with id 2 sent unhandled notification $/status/report:
{
  "text": "😺Lua",
  "tooltip": "Workspace   : /Users/spallen/.config/nvim-Modular\nCached files: 48/33\nMemory usage: 8M"
}
2024-03-02T19:10:27-05:00 [INFO] Language server with id 3 sent unhandled notification $/status/report:
{
  "text": "😺Lua",
  "tooltip": "Workspace   : /Users/spallen/.config/nvim-AstroNvimStart\nCached files: 72/69\nMemory usage: 27M"
}
2024-03-02T19:11:55-05:00 [INFO] Language server with id 2 sent unhandled notification $/status/report:
{
  "text": "😺Lua",
  "tooltip": "Workspace   : /Users/spallen/.config/nvim-Modular\nCached files: 48/33\nMemory usage: 9M"
}
```

**Comment by Moshyfawn** [comment]
Just for the record, the conversation started in #8551.

**Comment by smpallen99** [comment]
Yes, back on Feb 28. Today is the 2nd time. BTW, I have the 2 zed windows still open. I've moved back to nvim, but thought I would leave them for now.

**Comment by JosephTLyons** [comment]
@smpallen99, are you using splits at all in your zed windows? Is your setup similar at all to this:

https://github.com/zed-industries/zed/issues/5020

Just curious if these are similar issues or entirely different.

**Comment by smpallen99** [comment]
I just tried it on version `0.126.1` and I can't reproduce it again. Tried with a window with a split like the video in #5020 and with 2 windows (which I originally found the issue on) and no issue. 

Thanks for the help. I'm closing this issue.

**Comment by bitwalker** [comment]
This happened to me today, what I'm observing is that when in insert mode, all input goes to the window in which insert mode was entered. Switching windows, even if the window is minimized/backgrounded, makes no difference. However, if you exit to normal mode, you can start typing in other windows again.

TL;DR: it appears that insert mode claims some kind of exclusive ownership over the input across all open windows, and it isn't until you exit insert mode that things return to normal.

Super frustrating to encounter when you don't know this, but obviously easy enough to workaround once you do.

Hope that helps!

**Comment by bitwalker** [comment]
Well it seems I'm mistaken, because I am currently fighting with it and both windows are in normal mode, so 🤷. There doesn't seem to be any rhyme or reason to why it starts happening. I do wonder if there is some interaction with visual mode, because the last thing I did before it started this again was yank a region of code from visual mode.

**Comment by smpallen99** [comment]
I've been experiencing it a lot. I found that, when i'm in the window that's misbehaving, I click into another pane, or the project pane, then back again seems to fix the focus....

**Comment by smpallen99** [comment]
hey @bitwalker , our paths cross again 👍 Small world, eh

**Comment by bitwalker** [comment]
@JosephTLyons I've run into this several times, and I gotta be honest, it is absolutely infuriating. I'm happy to run various experiments and even try to contribute a fix if you have any ideas on where I can start. I'm working between several codebases in the same project right now, and bouncing between windows, and this happens basically every single time I switch windows, and is driving me nuts haha.

Anything I can do to help, I'm willing!
